src/coredump/coredump.c

```diff
@@ -1342,14 +1346,72 @@ static int gather_pid_metadata_from_argv(
                         t = buf;
                 }
 
+                if (i == META_ARGV_PID) {
+                        /* Store this so that we can check whether the core will be forwarded to a container
+                         * even when the kernel doesn't provide a PIDFD. Can be dropped once baseline is
+                         * >= v6.16. */
+                        pid_t kernel_pid;
+
+                        r = parse_pid(t, &kernel_pid);
+                        if (r < 0)
+                                return log_error_errno(r, "Failed to parse PID \"%s\": %m", t);
+
+                        r = pidref_set_pid(&kernel_pidref, kernel_pid);
```

Comment:
actually, we have pidref_set_pidstr()

Reply:
hmm, and if we have the pidref here anyway, i think we should maybe keep it around and initialize context->pidref from it too.